filebeat/prospector/config.go

```diff
@@ -27,24 +27,25 @@ var (
 )
 
 type prospectorConfig struct {
-	common.EventMetadata `config:",inline"`      // Fields and tags to add to events.
-	Enabled              bool                    `config:"enabled"`
-	DocumentType         string                  `config:"document_type"`
-	ExcludeFiles         []match.Matcher         `config:"exclude_files"`
-	IgnoreOlder          time.Duration           `config:"ignore_older"`
-	Paths                []string                `config:"paths"`
-	ScanFrequency        time.Duration           `config:"scan_frequency" validate:"min=0,nonzero"`
-	InputType            string                  `config:"input_type"`
-	CleanInactive        time.Duration           `config:"clean_inactive" validate:"min=0"`
-	CleanRemoved         bool                    `config:"clean_removed"`
-	HarvesterLimit       uint64                  `config:"harvester_limit" validate:"min=0"`
-	Symlinks             bool                    `config:"symlinks"`
-	TailFiles            bool                    `config:"tail_files"`
-	JSON                 *reader.JSONConfig      `config:"json"`
-	Pipeline             string                  `config:"pipeline"`
-	Module               string                  `config:"_module_name"`  // hidden option to set the module name
-	Fileset              string                  `config:"_fileset_name"` // hidden option to set the fileset name
-	Processors           processors.PluginConfig `config:"processors"`
+	common.EventMetadata   `config:",inline"`      // Fields and tags to add to events.
+	Enabled                bool                    `config:"enabled"`
+	DocumentType           string                  `config:"document_type"`
+	ExcludeFiles           []match.Matcher         `config:"exclude_files"`
+	IgnoreOlder            time.Duration           `config:"ignore_older"`
+	Paths                  []string                `config:"paths"`
+	ScanFrequency          time.Duration           `config:"scan_frequency" validate:"min=0,nonzero"`
+	InputType              string                  `config:"input_type"`
+	CleanInactive          time.Duration           `config:"clean_inactive" validate:"min=0"`
+	CleanRemoved           bool                    `config:"clean_removed"`
+	HarvesterLimit         uint64                  `config:"harvester_limit" validate:"min=0"`
+	Symlinks               bool                    `config:"symlinks"`
+	TailFiles              bool                    `config:"tail_files"`
+	JSON                   *reader.JSONConfig      `config:"json"`
+	Pipeline               string                  `config:"pipeline"`
+	Module                 string                  `config:"_module_name"`  // hidden option to set the module name
+	Fileset                string                  `config:"_fileset_name"` // hidden option to set the fileset name
+	Processors             processors.PluginConfig `config:"processors"`
+	DoubleWildcardMaxDepth uint8                   `config:"double_wildcard_max_depth"`
```

Comment:
* Where is the default set for this one? 
* The config name seems to be a bit long

I would like to have this feature disabled by default and people who need it enable it specifically. This also makes it easier for us to debug it in the future in case of errors, because we know someone enabled it. 

I like the idea of defining the max depth as in case someone just needs one or two levels, this would have positive performance implications.